### PR TITLE
add material grid

### LINKED ISSUE #1007: add material layout grid
As a designer, I'd like to be able to specify certain interface elements in terms of how many columns they use. Columns can shrink and grow as necessary for a responsive layout. 

As an engineer, I'd like to be able to apply a simple CSS style that will location of an element as specified by the designer. 

We will try applying [mdc-layout-grid](https://github.com/material-components/material-components-web/tree/master/packages/mdc-layout-grid) to see if it can give us a simple 12 column grid for desktop layout.

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

closes #1007

#### What's this PR do?

This PR redoes our layout to use the material grid for column widths, instead of our own custom stuff. I added two react components, `<Grid />` and `<Cell />`, which are basically just divs with the right classes set on them. These can be used to make a material grid wherever we want. I went ahead and replaced the existing layout in our `singleColumn` and `withChannelSidebar` components with this setup.

#### How should this be manually tested?

Basically everything should look the same throughout the app, and things should behave at all screenwidths and so on. Code should make sense.

### CODE REVIEW

File: static/js/components/Grid_test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        <div className="child" />
+      </Grid>
+    )
+    assert.equal(wrapper.props().className, "mdc-layout-grid")
+    assert.ok(wrapper.find(".mdc-layout-grid__inner").exists())
+    assert.ok(wrapper.find(".child").exists())
+  })
+
+  //
+  ;[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12].forEach(width => {
```
Comment (mbertrand) [inline static/js/components/Grid_test.js]: Minor nitpick: could probably just test values `0` and `12` for `width` here, test passes no matter what `width` is: 0-12, or 99, or 'foo'.  It's just Flow that will complain about the bad values.

Comment (aliceriot) [reply static/js/components/Grid_test.js]: yeah, I had added this mainly because the test file is checked with flow, but I can change it to just check 1 and 12
